var/spack/repos/builtin/packages/py-pyglet/package.py

```diff
@@ -11,12 +13,30 @@ class PyPyglet(PythonPackage):
     for developing games and other visually rich applications.
     """
 
-    homepage = "https://github.com/pyglet/pygle://github.com/pyglet/pyglet"
-    pypi = "pyglet/pyglet-1.4.2.tar.gz"
+    homepage = "https://github.com/pyglet/pyglet"
+    pypi = "pyglet/pyglet-2.0.9.zip"
 
+    version("2.0.10", sha256="242beb1b3bd67c5bebdfe5ba11ec56b696ad86b50c6e7f2a317f8d783256b9c9")
+    version("2.0.9", sha256="a0922e42f2d258505678e2f4a355c5476c1a6352c3f3a37754042ddb7e7cf72f")
     version("1.4.2", sha256="fda25ae5e99057f05bd339ea7972196d2f44e6fe8fb210951ab01f6609cdbdb7")
     version("1.2.1", sha256="d1afb253d6de230e73698377566da333ef42e1c82190216aa7a0c1b729d6ff4d")
 
     depends_on("py-setuptools", type="build")
-    depends_on("py-future", type=("build", "run"))
-    depends_on("python@2.7:2.8,3.4:", type=("build", "run"))
+    depends_on("py-future", type=("build", "run"), when="@:1")
+    depends_on("python@2.7:2.8,3.4:", type=("build", "run"), when="@:1")
+    depends_on("python@3.8:", type=("build", "run"), when="@2.0:")
+    depends_on("gl", type=("build", "run"), when="@2.0: platform=linux")
```

Comment:
Seems like GLX is required too.

Reply:
I guess I misunderstood something in the remarks above?

https://github.com/spack/spack/pull/41082#discussion_r1401931097  
https://github.com/spack/spack/pull/41082#discussion_r1403394006

btw, *is* there even a `gl` package?

```
» spack info gl
==> Error: Package 'spack.pkg.builtin.gl' not found.
Use 'spack create' to create a new package.

Did you mean one of the following packages?
  wgl
  gsl
  glx
```

Reply:
@adamjstewart I thought we should use virtual packages where possible.

@heerener you can see which providers are available for a virtual package with `spack providers gl`

Reply:
Ah, we used to have gl and glx virtual deps, but they all got refactored into a more complicated hierarchy. Sorry for the confusion!

Reply:
Thanks, all of this makes more sense now :-)